### PR TITLE
DT-4933 // When arg expansion fails, use untouched arg

### PR DESCRIPTION
**Pull request by pballandras**
## Context
In the util package, we used to try and expand args when certain conditions where met, but those conditions could be met without the arg being expandable. This lead to args simply disappearing out of the command and thus, the command being wrongly executed. 

## Now
Now, we still try to expand the arg, but when that fails, we recover the original arg and use it untouched. I also added very minimal tests on the function since it was not tested. If any of the reviewer want to add a test case, I'll gladly add it since we never have enough cases.

### CODE REVIEW

File: util/file_test.go [hunk 1]
```diff
@@ -89,3 +89,19 @@ func TestPathContainsHiddenFileOrFolder(t *testing.T) {
...
+	testCases := []struct {
+		args     []interface{}
+		expected []interface{}
+	}{
+		{[]interface{}{"1"}, []interface{}{"1"}},
+		{[]interface{}{"words"}, []interface{}{"words"}},
+		{[]interface{}{"something.something\\[0\\]"}, []interface{}{"something.something\\[0\\]"}},
+		{[]interface{}{"-target"}, []interface{}{"-target"}},
+		{[]interface{}{"-lock-timeout=20m"}, []interface{}{"-lock-timeout=20m"}},
+	}
```
[the remark this answers is not in the document]
Comment (pballandras) [reply util/file_test.go:103]: I wanted to add tests because there were none, but I don't have use cases on the top of my head of expandable args, so I just added those that should not expand. If you have examples I'd be glad to add them.

> Also, if the supplied original arg include the current path, it will be changed even if there is no replacement.

Good catch, I'll make adjustments.

[the remark this answers is not in the document]
Comment (jocgir) [reply util/file_test.go:103]: You may create a test fixture folder (or use an existing one) and add some tests with more than one parameter that use wildcards and expand to some files.

[the remark this answers is not in the document]
Comment (pballandras) [reply util/file_test.go:106]: I added an env expansion just for good mesure ;)

**Review by jonapich — APPROVED** [review]

**Review by jocgir — APPROVED** [review]
